torch/utils/data/sampler.py

```diff
@@ -12,24 +12,46 @@
     "WeightedRandomSampler",
 ]
 
-T_co = TypeVar('T_co', covariant=True)
+T_co = TypeVar('T_co', int, List[int], covariant=True)
 
 
 class Sampler(Generic[T_co]):
     r"""Base class for all Samplers.
 
     Every Sampler subclass has to provide an :meth:`__iter__` method, providing a
-    way to iterate over indices of dataset elements, and a :meth:`__len__` method
+    way to iterate over indices or lists of indices (batches) of dataset elements, and a :meth:`__len__` method
     that returns the length of the returned iterators.
 
+    Example:
+        class AccedingSequenceLengthSampler(Sampler[int]):
+            def __init__(self, data: List[str]):
+                self.data = data
+
+            def __len__(self):
+                return len(self.data)
+
+            def __iter__(self):
+                sizes = torch.tensor([len(x) for x in self.data])
+                yield from torch.argsort(sizes).tolist()
```

Comment:
nit: add return types for each method in the examples

Reply:
done